src/storage/Serializer.h

```diff
@@ -2,14 +2,48 @@
 
 #pragma once
 
+#include <string>
+#include <string_view>
+
 #include "zeek/Val.h"
 
 namespace zeek::storage {
 
+/**
+ * Base class for a serializer used by storage backends.
+ */
 class Serializer {
 public:
+    /**
+     * A function pointer to a string transformation function that backends can use to
+     * force formatting on the serializer output.
+     */
+    using string_transform = std::function<std::string(std::string_view)>;
```

Comment:
Wouldn't it be more consistent to have this take a byte_span and return a byte_buffer?

But also, I'm not sure we should have it. If it's added as a parameter to `Serialize()`, all serializers need to implement the same logic. To me it feels more like backends that need to sanitize/munge the serializer output should do so explicitly, instead of pushing the transformer logic down into `Serialize()`:
```
// not sure this works, but something like it:
serializer->Serialize(val).and_then(string_transform).or_else(blow_up)
```

Reply:
It was an experiment for sure. In the end I'm not the biggest fan of it either. I think the only problem it solves is preventing the backends from having to convert back from bytes to do their transformation. For example, the NATS backend converts any non-alnum character into byte-strings (`x00`-style). If this happens after the call to `Serialize()`, it's going to need convert back to a string from the byte_buffer result and then do the transformation.

Reply:
I'm going to just drop out that entire commit. I don't see it being useful enough to keep as a generic thing in the serializers.